### PR TITLE
fix(instrumentation): propagate correct session.id to Langfuse via langfuse.session.id

### PR DESCRIPTION
**Pull request by MrOrz**
## Problem

The current Langfuse integration would generate multiple Langfuse sessions for 1 chatroom.

Ex: 
- the main session https://langfuse.cofacts.tw/project/cmm0emerr0001qi07eugd0760/sessions/6498b3f2-860c-494f-a955-5461688c06d6
- a trace with correct session_id but somehow created wrong Langfuse session https://langfuse.cofacts.tw/project/cmm0emerr0001qi07eugd0760/sessions/5eaedb9e-12b9-4664-93e6-41352bedf39d

## Summary

- Adds `RootSessionSpanProcessor` to ensure all spans in a trace land in the correct Langfuse session
- `openinference-instrumentation-google-adk` incorrectly stamps sub-agent spans with ADK-internal session UUIDs instead of the user-facing `session_id` (upstream bug: https://github.com/Arize-ai/openinference/issues/3117)
- Fixes this by reading the parent span's session at `on_start` time and writing it as `langfuse.session.id`, which Langfuse gives precedence over `session.id` in OTLP ingestion
- Stateless implementation — no dicts, no cleanup logic, no edge cases

## Test plan

Subagent spans are now assigned with parent session ID
圖片

- [x] Trigger an invocation that calls a sub-agent (investigator or verifier)
- [x] Confirm the Langfuse trace `sessionId` matches the user-facing ADK session shown in the chat URL
- [x] Confirm all observations (including `call_llm` spans under sub-agents) carry `langfuse.session.id` equal to the root session

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: adk/cofacts_ai/instrumentation.py [hunk 1]
```diff
@@ -1,13 +1,65 @@
...
+            # Root or context-detached sub-agent span: register the first one seen.
+            # setdefault is atomic under the GIL and returns the winning value.
+            root = self._root_sessions.setdefault(trace_id, session_id)
+            if root != session_id:
+                span.set_attribute(_SESSION_ID_ATTR, root)
+            return
+
+        root = self._root_sessions.get(trace_id)
+        if root and session_id != root:
+            span.set_attribute(_SESSION_ID_ATTR, root)
```
[the remark this answers is not in the document]
Comment (MrOrz) [reply adk/cofacts_ai/instrumentation.py]: Rewritten, this no longer applies

File: adk/cofacts_ai/instrumentation.py [hunk 2]
```diff
@@ -20,6 +72,9 @@ def setup_instrumentation():
 
     if langfuse.auth_check():
         GoogleADKInstrumentor().instrument()
+        cast(SDKTracerProvider, otel_trace.get_tracer_provider()).add_span_processor(
+            RootSessionSpanProcessor()
+        )
```
[the remark this answers is not in the document]
Comment (MrOrz) [reply adk/cofacts_ai/instrumentation.py:70]: Rewritten, this no longer applies

File: adk/cofacts_ai/instrumentation.py [hunk 3]
```diff
@@ -1,13 +1,65 @@
...
+                span.set_attribute(_SESSION_ID_ATTR, root)
+            return
+
+        root = self._root_sessions.get(trace_id)
+        if root and session_id != root:
+            span.set_attribute(_SESSION_ID_ATTR, root)
+
+    def on_end(self, span):
+        if span.parent is None and span.context is not None:
+            self._root_sessions.pop(span.context.trace_id, None)
```
[the remark this answers is not in the document]
Comment (MrOrz) [reply adk/cofacts_ai/instrumentation.py]: no longer valid

**Review by nonumpa — APPROVED** [review]
